cdap-common/src/main/java/co/cask/cdap/gateway/handlers/StackHandler.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+/**
+ * Returns information about threads, including their stack traces.
+ */
+public class StackHandler extends AbstractHttpHandler {
+
+  private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
+  private static final int STACK_DEPTH = 20;
+
+  @Path(Constants.Gateway.API_VERSION_3 + "/system/services/{service-name}/stacks")
+  @GET
+  public void stacks(HttpRequest request, HttpResponder responder) {
+    // ignore the service-name, since we dont need it. its only used for routing
```

Comment:
It'll be good to include the service name in the output as well, so that by looking at the output, it's easier to tell what service it is.